packages/tools/src/stateManagement/segmentation/SegmentationStateManager.ts

```diff
@@ -277,26 +346,21 @@ export default class SegmentationStateManager {
         );
         const segImage = cache.getImage(imageIds[0]);
         if (segImage?.FrameOfReferenceUID === frameOfReferenceUID) {
-          convertStackToVolumeSegmentation(segmentation);
+          internalConvertStackToVolumeLabelmap(segmentation);
         }
       } else {
         // TODO: Implement Volume Labelmap on Volume Viewport
       }
     }
-
-    // make all the other representations inactive first
-    this.setActiveSegmentationRepresentation(
-      viewportId,
-      segmentationRepresentationUID
-    );
   }
 
   /**
    * Helper function to update labelmap segmentation image references.
-   * @param segmentationId - The ID of the segmentation representation.
-   * @param stackViewport - The stack viewport.
-   * @param labelmapImageIds - The labelmap image IDs.
-   * @param updateCallback - A callback to update the reference map.
+   * @param {string} segmentationId - The ID of the segmentation representation.
+   * @param {Types.IStackViewport} stackViewport - The stack viewport.
+   * @param {string[]} labelmapImageIds - The labelmap image IDs.
+   * @param {Function} updateCallback - A callback to update the reference map.
+   * @returns {string | undefined} The labelmap imageId reference for the current imageId rendered on the viewport.
    */
   _updateLabelmapSegmentationReferences(
     segmentationId,
```

Comment:
This may well not depend on it being a stack viewport any longer - is an IViewport sufficient?  
For the update callback, could you type that with function arguments?